docs/atoz.md

```diff
@@ -0,0 +1,487 @@
+
+### Required dependencies
+
+You will need the package `build-essential` to be installed. We need this specifically for `node-gyp` to build packages when using `npm` or `yarn` to install packages.
+
+As the `root` user run these commands:
+
+~~~bash
+apt update
+apt install -y build-essential
+~~~
+
+If you do not have root access and just want to check the dependency is installed you can use this command:
+
+~~~bash
+dpkg -s build-essential | grep Status:
+~~~
+
+If the program is installed you will see this result:
+
+~~~bash
+Status: install ok installed
+~~~
+
+### Create a local user account
+
+For this guide, unless specifically stated, you should not use a `root` account to install and run `wetty`. Please use an existing local account or create one now. 
+
+**Note:** Whichever user runs `wetty` should be the same user you wish to authenticate with via `ssh` to keep this simple.
+
+If you need to create a local user account you can run this command: 
+
+**Important note:** replace `username` with a user name of your choosing and create a password when prompted
+
+~~~bash
+adduser --gecos "" username
+~~~
+
+Switch to your local user now and open an `ssh` session to continue with this guide.
+
+### Install node locally
+
+To install and manage `node` as a local user we are going to use [Node Version Manager](https://github.com/nvm-sh/nvm) as an established solution to installing and managing multiple versions of node without needing `root` access. We are going to install the `lts` or long term support release of `node` to use with this application.
+
+~~~bash
+bash <(curl -s https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh) && source ~/.profile
+nvm install --lts
+~~~
+
+You can now call `node` to check it works using this command. 
+
+~~~bash
+node -v
+~~~
+
+Your result should look something like this.
+
+~~~bash
+v12.16.2
+~~~
+
+**Note:** There is consideration with this method. `node` is only in the local user's path through sourcing of the `~/.nvm/nvm.sh` via the users `.bashrc` file. Unless this is done `node` will not be usable unless directly linked to and `nvm` commands will be unavailable. 
+
+The way we over come this issue for the needs of this guide is by using this command where applicable:
+
+~~~bash
+source ~/.nvm/nvm.sh && nvm which 12
+~~~
+
+**Why?** This command will always provide us with the path to the most current version of `node 12` installed via `nvm` regardless of other versions of `node` installed.
+
+### Generate openssl certificates.
+
+**Why?** So that later we can configure `wetty` to work with `https` and make sure we interact with `wetty` over a secure connection at all times.
```

Comment:
also plz link to letencrypt as users may not want to self sign

Reply:
The current guide assumed a vps + IP as the host platform with no domain or nginx. As far as i am aware lets encrypt does not currently support issuing certs for a [public IP](https://letsencrypt.org/upcoming-features/#ip-addresses-in-certificates)?

So literally spin ups a VPS and copy and paste and you'll have the access to `wetty` over https.

To use letsencrypt I'd assume they have a valid domain and are using nginx, then this ssl method would sit behind that proxy pass and their valid cert anyway?

Like this:

`wetty` > `self-signed ssl` > `browser` via `https://IP:PORT`

`wetty` > `self-signed ssl` > `nginx proxy` > `letsencrypt ssl` > `browser` via `https://DOMAINorIP/wetty`

I can add it but maybe it would be better suited in the nginx section?